src/infra/dex/paraswap/mod.rs

```diff
@@ -50,73 +50,41 @@ impl ParaSwap {
         }
     }
 
+    /// Make a request to the `/swap` endpoint.
     pub async fn swap(
         &self,
         order: &dex::Order,
-        slippage: &dex::Slippage,
+        slippage: dex::Slippage,
         tokens: &auction::Tokens,
     ) -> Result<dex::Swap, Error> {
-        let price = self.price(order, tokens).await?;
-        let transaction = self.transaction(&price, order, tokens, slippage).await?;
+        let query = dto::SwapQuery::new(&self.config, order, tokens, slippage)?;
+        let swap = util::http::roundtrip!(
```

Comment:
why not to make it into a funcion as `price()` was?

Reply:
Do you mean dto into domain? Because it also requires an `order` instance for the domain object to be constructed.

Reply:
I think price and transaction functions were separated because they called different logical endpoints. We now have a single endpoint and I don't think it makes sense to extract it since it will be used just in a single place and doesn't have a lot of lines of code.